File: KazNTU.API/KazNTU.API/KazNTU.API/Models/Department.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Web;

namespace KazNTU.API.Models
{
    public class Department
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }
        public string Title { get; set; }

        public ICollection<User> Users { get; set; }
    }
}

File: KazNTU.WEB/libs/backend/src/models/department.ts
import { User } from './user';

export class Department {
    public Id: string;
    public Title: string;
    public Users: User[];

    constructor(...parts: Partial<Department>[]) {
        Object.assign(this, ...parts);
        if (this.Users) {
            this.Users = this.Users.map(u => new User(u));
        }
    }
}